xformers/ops/memory_efficient_attention.py

```diff
@@ -733,16 +822,71 @@ def memory_efficient_attention(
     attn_bias: Optional[Union[torch.Tensor, AttentionMask]] = None,
     p: float = 0.0,
     *,
-    op=None,
-):
-    """
-    Implements the memory-efficient attention mechanism following
+    op: Optional[AttentionOp] = None,
+) -> torch.Tensor:
+    """Implements the memory-efficient attention mechanism following
```

Comment:
super comprehensive and on point, thanks @danthe3rd. I tend to favor doc in the code, doesn't rot as fast, and this looks as good as it gets